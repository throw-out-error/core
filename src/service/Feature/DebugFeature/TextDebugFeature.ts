import { FF } from "../../FeatureFactory";
import { KernelServices } from "../KernelFeature";
import { drawDependencies } from "./drawDependency";

export interface DebugFeatureServices {
    getDependencyTree(): string;
}

export const TextDebugFeature: FF<
    DebugFeatureServices,
    KernelServices
> = function textDebugFeature({ kernel }) {
    return {
        getDependencyTree: () => () => drawDependencies(kernel().references()),
    };
};
