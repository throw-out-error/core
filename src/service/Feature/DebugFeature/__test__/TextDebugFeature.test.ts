import triviality, { FF } from "../../../index";
import { TextDebugFeature } from "../TextDebugFeature";

it("Without unknown services", async () => {
    const dependencyContainer = await triviality()
        .add(TextDebugFeature)
        .build();
    expect(dependencyContainer.getDependencyTree()).toMatchSnapshot();
});

interface Services {
    a: string;

    b: string;

    c: string;

    abc: string;

    nested: string;
}

const ComplexFeature: FF<Services> = function CF({ services }) {
    const [a, b, c, abc] = services("a", "b", "c", "abc");

    return {
        a: () => "a",
        b: () => "b",
        c: () => "c",
        single: () => a(),
        ab: () => a() + b(),
        abc: () => a() + b() + c(),
        nested: () => abc(),
    };
};

it("With complex dependencies", async () => {
    const dependencyContainer = await triviality()
        .add(TextDebugFeature)
        .add(ComplexFeature)
        .build();
    expect(dependencyContainer.getDependencyTree()).toMatchSnapshot();
});
