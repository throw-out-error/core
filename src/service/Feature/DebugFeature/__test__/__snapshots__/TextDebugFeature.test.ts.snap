// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With complex dependencies 1`] = `
"              ________________________________________________________________________________________________________________________________________________________________________________
             /                           \\\\                                  \\\\                       \\\\        \\\\        \\\\           \\\\             \\\\                   \\\\                         \\\\
  #1 kernelFeature.kernel  #3 setupFeature.setupCallbacks  #4 textDebugFeature.getDependencyTree  #5 CF.a  #6 CF.b  #7 CF.c  #8 CF.single    #9 CF.ab           #10 CF.abc_________      #11 CF.nested
                                                                                                                                   |        /        \\\\         /          \\\\        \\\\           |
                                                                                                                                #5 CF.a  #5 CF.a   #6 CF.b  #5 CF.a     #6 CF.b  #7 CF.c  #10 CF.abc_________
                                                                                                                                                                                         /          \\\\        \\\\
                                                                                                                                                                                      #5 CF.a     #6 CF.b  #7 CF.c"
`;

exports[`Without unknown services 1`] = `
"              ______________________________________________________________
             /                           \\\\                                  \\\\
  #1 kernelFeature.kernel  #3 setupFeature.setupCallbacks  #4 textDebugFeature.getDependencyTree"
`;
